utils/staking-miner/src/main.rs

```diff
@@ -0,0 +1,651 @@
+							let extrinsic = UncheckedExtrinsic::new_signed(call, address, signature.into(), extra);
+							log::debug!(
+								target: crate::LOG_TARGET, "constructed extrinsic {}",
+								sp_core::hexdisplay::HexDisplay::from(&extrinsic.encode())
+							);
+							extrinsic
+						}
+					}
+				}
+			}
+	};
+}
+
+// TODO: we might be able to use some code from the bridges repo here.
+fn signed_ext_builder_polkadot(
+	nonce: Index,
+	tip: Balance,
+	era: sp_runtime::generic::Era,
+) -> polkadot_runtime_exports::SignedExtra {
+	use polkadot_runtime_exports::Runtime;
+	(
+		frame_system::CheckSpecVersion::<Runtime>::new(),
+		frame_system::CheckTxVersion::<Runtime>::new(),
+		frame_system::CheckGenesis::<Runtime>::new(),
+		frame_system::CheckMortality::<Runtime>::from(era),
+		frame_system::CheckNonce::<Runtime>::from(nonce),
+		frame_system::CheckWeight::<Runtime>::new(),
+		pallet_transaction_payment::ChargeTransactionPayment::<Runtime>::from(tip),
+		runtime_common::claims::PrevalidateAttests::<Runtime>::new(),
+	)
+}
+
+fn signed_ext_builder_kusama(
+	nonce: Index,
+	tip: Balance,
+	era: sp_runtime::generic::Era,
+) -> kusama_runtime_exports::SignedExtra {
+	use kusama_runtime_exports::Runtime;
+	(
+		frame_system::CheckSpecVersion::<Runtime>::new(),
+		frame_system::CheckTxVersion::<Runtime>::new(),
+		frame_system::CheckGenesis::<Runtime>::new(),
+		frame_system::CheckMortality::<Runtime>::from(era),
+		frame_system::CheckNonce::<Runtime>::from(nonce),
+		frame_system::CheckWeight::<Runtime>::new(),
+		pallet_transaction_payment::ChargeTransactionPayment::<Runtime>::from(tip),
+	)
+}
+
+fn signed_ext_builder_westend(
+	nonce: Index,
+	tip: Balance,
+	era: sp_runtime::generic::Era,
+) -> westend_runtime_exports::SignedExtra {
+	use westend_runtime_exports::Runtime;
+	(
+		frame_system::CheckSpecVersion::<Runtime>::new(),
+		frame_system::CheckTxVersion::<Runtime>::new(),
+		frame_system::CheckGenesis::<Runtime>::new(),
+		frame_system::CheckMortality::<Runtime>::from(era),
+		frame_system::CheckNonce::<Runtime>::from(nonce),
+		frame_system::CheckWeight::<Runtime>::new(),
+		pallet_transaction_payment::ChargeTransactionPayment::<Runtime>::from(tip),
+	)
+}
+
+construct_runtime_prelude!(polkadot, NposCompactSolution16);
+construct_runtime_prelude!(kusama, NposCompactSolution24);
+construct_runtime_prelude!(westend, NposCompactSolution16);
+
+#[macro_export]
+macro_rules! any_runtime {
+	($($code:tt)*) => {
+		unsafe {
+			match $crate::RUNTIME {
+				$crate::AnyRuntime::Polkadot => {
+					use $crate::polkadot_runtime_exports::*;
+					$($code)*
+				},
+				$crate::AnyRuntime::Kusama => {
+					use $crate::kusama_runtime_exports::*;
+					$($code)*
+				},
+				$crate::AnyRuntime::Westend => {
+					use $crate::westend_runtime_exports::*;
+					$($code)*
+				}
+			}
+		}
+	}
+}
+
+#[derive(codec::Encode, codec::Decode, Clone, Copy, Debug)]
+#[allow(unused)]
```

Comment:
All of this is just some ideas... I might remove it later, but I think it can stay for now. It pretty much demonstrates what the miner should do in case of failuer.